cache/in-memory/src/permission.rs

```diff
@@ -361,7 +361,11 @@ impl<'a> InMemoryCachePermissions<'a> {
         let calculator =
             PermissionCalculator::new(guild_id, user_id, everyone, assigned.as_slice());
 
-        Ok(calculator.in_channel(channel.kind, overwrites.as_slice()))
+        let permissions = calculator.in_channel(channel.kind, overwrites.as_slice());
+
+        Ok(self
+            .communication_disabled_perms(permissions, user_id, guild_id)
+            .unwrap_or(permissions))
```

Comment:
These can use `unwrap_or_else` and call the above function lazily instead of eagerly

Reply:
i’m not sure i understand, the permissions are required either way